### ISSUE TITLE
WavesFX 2 - [Track 2. Web3.0 Development Grant] 

### ISSUE THREAD

**Issue by suadzh** [Waves-Association/grants-program#13]
- **Proposer:** 
Suad Hasani
- **Contacts:**
suadzh@protonmail.com
- **Github username:**
https://github.com/suadzh

## 👥  Team 

- Are you applying to implement the grant as an individual developer or do you represent a team?
I am the sole developer of the project.

- What’s your team or startup name? What is a website? (in case applying as a team or startup)
WavesFX
https://wavesfx.github.io/

## 🔍 Overview

Please describe your solution idea by the following structure:

**Short description of the solution you suggest**
WavesFX is an open-source and cross-platform desktop wallet for the Waves Protocol. The first version of WavesFX was released in December 2019 and developed throughout 2020. WavesFX is now in the development phase of the 2.0 version.

**How does it solve the specified problem?**
Apart from all essential functions (transfer, mass transfer, leasing, burning, token issue/reissue and alias creation etc), the first version of WavesFX already offered a wide range of advanced features like transaction batching and an offline mode for cold wallets.

With the introduction of Neutrino, Waves has made big steps towards DeFi/DeFo which led to smart contracts being used on a daily basis. Currently, interacting with dApps/smart contracts is rather difficult for users with no programming skills unless dApps provide a specific GUI. WavesFX 2 will provide an easy way for users to interact with any dApp via a graphical user interface. 

In addition to new features, the 2.0 release also includes a new code base and a revamped/modernized GUI.

**How you will integrate this project into Waves / Gravity / Neutrino.**
General use cases like transferring assets, leasing etc. obviously still exist but the main difference is that users will be able to interact with smart contracts. WavesFX 2.0 will include a preset manager for dApps. Neutrino will be one of the default presets.

**Why your team is interested in creating this project**
A rich ecosystem is important for any platform. The largest blockchain ecosystems provide a vast range of open source toolings, libraries for all kinds of programming languages but also a variety of wallets. Some of these wallets are based on different technologies, provide additional functionality and others aim to slim down functionality in order to provide a more lightweight experience. As mentioned above, WavesFX tackles both aspects and having fallback options is important.

**What is the target audience of the final solution?**
WavesFX targets both beginners and advanced users.

**What is the “secret sauce” of it?**
WavesFX combines elements familiar to wallet users from other blockchain platforms with Waves' powerful capabilites without sacrificing availability or performance. Regular users can use WavesFX as their main wallet and developers can use WavesFX for development, basically a wallet for everyone.

**What is the duration of the whole project?**
While it is more of an ongoing process, 2.x would likely mean a year until a 3.0 version would possibly be released in 2022.

**Resources required to implement the grant task**
Development, UI/UX design, maintenance/fixing, implementation of new features, app signing, documentation & translation.

### Ecosystem Fit

**Are there any other projects similar to yours? If so, how is your project different?**
WavesFX is the only open-source Waves desktop wallet with full-fledged functionality. Waves Exchange rather stands as the flagship of the Waves Protocol and puts a bigger focus on DeFi/DeFo and trading functionality. Waves Exchange is also a web application with a large number of users, which practically forces them to keep the source code private in order to prevent phishing attacks and SIGN serves a very specific use case. WavesFX could best be described as the Electrum of the Waves Protocol.

## 🔩 Development Roadmap 

This section should break out the development roadmap into a number of milestones. It helps to describe the functionality we should expect, plus how we can check that such functionality exists in the product. Whenever milestones are delivered, we refer to the roadmap to ensure that everything has been delivered as expected.

**WavesFX 2.0 will be released in January or early Februrary 2021. Subsequently, feature releases will follow.**

### Overview

- **Total Estimated Duration:** Duration of the whole project
WavesFX 2 will be continuously developed in 2021.
- **Total Costs:** Amount of Payment in Waves for the whole project. The total amount of funding needs to not more than 100k Waves.
10 000 Waves

## Additional Information ➕

Any additional information that you think is relevant to this application that hasn't already been included.

Possible additional information to include:

Landing: https://wavesfx.github.io/
Git Repository: https://github.com/wavesfx
Twitter: https://twitter.com/wavesfxwallet

**Comment by KardanovIR** [comment]
Hey, Waves Association Grants & Funding working group reviewed your application and recommended your project to the Waves Association board.

The board will make the final decision, but the working group also provides the next feedback and recommendations.


**Feedback:**

- WavesFX supports minor protocol features that are not in the Waves Exchange wallet
- The only open-source wallet in the ecosystem now
- It’s cross-platform and runs everywhere where Java runs, which makes it more accessible for some categories of users
- Ledger support will make the wallet the most secure Waves wallet
- Anonymous transactions based on zk-SNARKs will bring new categories of users into the Waves ecosystem

**Recommendations:**

- Add expenses breakdown and milestones

**Comment by suadzh** [comment]
**WavesFX 2 (Phase) Report - September 2021**

WavesFX 2 was originally planned to be released in Q1/Q2 but delayed to Q4.

- First iteration of core implementation was finished in Q1
- GUI still in the works due to a complete overhaul of the initial design which caused codebase inconsistencies and performance issues for which FXML is largely to blame. 
- It has been decided to **rebrand WavesFX completely**. After its next major release, WavesFX will be called **Phase**. The two main reasons for the change are firstly that the name WavesFX (derived from the platform JavaFX) conflicts with eventual ports to mobile platforms (which most certainly would not be based on JavaFX) in the future, and secondly, that the new name and branding provides better marketability.
- As partially mentioned above, the look and theme of the wallet is still under revision as a result of the rebranding.
- It is still up for decision whether there should be separate sections in the wallet for dedicated DeFi applications such as Swop.fi or Neutrino. 
Whilst dedicated sections for dApps would certainly be more user friendly, the idea behind leaving out such sections is that Phase should act as an independent part of a decentralized platform since users will be able to communicated with dApps via the wallet's smart contract interface anyway. Hard-coding third-party dApps into the GUI would lead to both technical and political dependence to dApps.

**Summary**

Putting aside the undesirable delay, everything is going according to plan and along with the wallet, the new landing page and documentations will all be released in Q4. New features that are already in development as well will be further developed and released in 2021.

**Comment by BlackTurtle123** [comment]
Hello @suadzh do you have a new update for us?

**Comment by BlackTurtle123** [comment]
> Hello @suadzh do you have a new update for us?

Hello!
Any update? :)

**Comment by suadzh** [comment]
> > Hello @suadzh do you have a new update for us?
> 
> Hello! Any update? :)

Yes! Pretty much finished. Just a bit of GUI and deployment flickering left. Will release in the coming weeks!
Sorry for the delay.

**Comment by BlackTurtle123** [comment]
Hello @suadzh any update?

**Comment by suadzh** [comment]
> Hello @suadzh any update?
Still working on it. Further details will be discussed with the Waves Association.

**Comment by suadzh** [comment]
> > > Hello @suadzh any update?
> > > Still working on it. Further details will be discussed with the Waves Association.
> 
> Hello, Can you give us a public update for the whole association for transparency It's been a year since the last public update

At the moment, I'm rewriting much of the core and going back to JSON from GRPC due to strange inconsistencies (for example, different PubKey sizes. Why?) in terms of protobuf schemes and RPC endpoints that seem to be different even in different block ranges. I've faced too many surprises during testing and decided to go back to what's proven to be solid. The minor upsides of GRPC in this use case aren't worth it.
Not much work left. Release is guaranteed this year.
